speechbrain/lobes/models/transformer/TransformerASR.py

```diff
@@ -224,6 +224,10 @@ class TransformerASR(TransformerInterface):
     use_linear_after_conv: bool, optional
         If True, will apply a linear transformation of size input_size//2.
         -> Branchformer
+    output_hidden_states: bool, optional
```

Comment:
@asumagic this PR adds the possibility to return the hidden states of ANY transformer model. I find it important as SSL evaluation requires an access to these states. What do you think of the implementation in this PR?

Reply:
The idea seems ok to me, though the docs should be more explicit on return type.